csrc/kernel.cpp

```diff
@@ -118,6 +118,19 @@ class KernelIrScanner : private IrVisitor {
         summary_.has_block_welford || out_dom->hasBlockReduction();
   }
 
+  void handle(GroupedReductionOp* grouped_rop) final {
+    summary_.has_iter_grouped_reductions = true;
+    int num_grouped_iterations = 1;
+    auto out_tv = ir_utils::getTvOutput(grouped_rop);
+    for (auto axis : out_tv->getLeafDomain()) {
+      if (axis->getParallelType() == ParallelType::Group) {
+        num_grouped_iterations *= (int)axis->extent()->value();
+      }
+    }
+    summary_.num_grouped_iterations =
```

Comment:
so `num_grouped_iterations` could still be 1 when we have `has_iter_gropued_reduction`?

Reply:
it can't be 1 if we have `has_iter_gropued_reduction` becuase we only group if vectorized and only vectorized if vectorization factor is larger than 1.
added check:
```
    NVF_ERROR(
        num_grouped_iterations == 2 || num_grouped_iterations == 4 ||
            num_grouped_iterations == 8 || num_grouped_iterations == 16,
        "Iteration grouped reduction only support grouping 2, 4, 8, or 16 iterations, but found ",
        num_grouped_iterations);
```

Reply:
I'm fine with it as-is for now.

But should we have maybe add another operation with `IterGroupedReductionOp` instead?

Reply:
resolved offline. we decided to keep it as-is in case we would want a future hybrid mode.